Newsletters: add 335 (2025-01-03) (#2111)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2025-01-03-newsletter.md

  allow the centralized coordinator to determine which user received
  which outputs, eliminating any benefit of using a sophisticated
  centralized coinjoin protocol.  Kogman provides evidence that
  the vulnerabilities were known to several wallet developers for years.
  A similar vulnerability affecting some of the same software was
  previously mentioned in [Newsletter #333][news333 vuln].

- **Updated ChillDKG draft:** Tim Ruffing and Jonas Nick [posted][rn
  chilldkg] to the Bitcoin-Dev mailing list a link to the [current draft
  BIP for ChillDKG][bip-chilldkg], which describes a distributed key
  generation protocol compatible with FROST [scriptless threshold
  signatures][topic threshold signature] for Bitcoin.  Since their
  initial announcement (see [Newsletter #312][news312 chilldkg]), they've
  fixed a security vulnerability, added an investigation phase that
  makes it possible to identify faulty participants, and made backup and
  recovery easier.  They've also worked with Sivaram Dhakshinamoorthy to
  keep their proposal in sync with his proposal for Bitcoin-compatible
  FROST signing (see [Newsletter #315][news315 frost]).

> ## Changing consensus

Reviewer: I am not sure whether this question makes sense in English, but is the “changing” in the title meant as adjective or as verb?

Or in other words, which is closer to the intent:

- section about how we are changing Bitcoin consensus
- section about Bitcoin consensus that is in the middle of a change

If it makes no sense, please just ignore :-)

Author: Talked to ChatGPT about this and I think it's the figure of speech known as _ellipsis_.  A full construction would be something like: "News about ideas for changing consensus rules", but I elide the parts that are likely to be understood: "News about ideas for" ('cause this is a newsletter) and "rules" (following longstanding convention of calling "consensus rules" just "consensus" for convenience).

It feels natural to me as a native speaker and seems in line with other stuff, e.g. the "Scaling Bitcoin" conference comes to mind (probably short for "Conference about ideas for scaling Bitcoin transactional throughput").  However, if you think it'll be too confusing to non-native speakers, let me know and we can use a different section title going forward.

Reviewer: Thanks. The full construct “News about ideas for changing consensus rules” answers my question.

It is not too confusing. Even though I can interpret the title in two different ways, the difference is subtle and I don't think it makes sense to spend time on clarifying it.

However, I wanted to better understand before I decide how to translate the title. In our language I cannot elide that much from the full construct.

Reviewer 2: The translation doesn't have to be exactly literal, for example "Změny konsenzu" would work just fine. (We can discuss this in the translation PR later.)

_A new monthly section summarizing proposals and discussion about
changing Bitcoin's consensus rules._

> - **CTV enhancement opcodes:** developer moonsettler posted to both

Author: > We insert the title "developer" out of necessity to prevent confusion between 0xB10C's nym and conflicting phrasing (e.g. a hex string).  -- https://github.com/bitcoinops/bitcoinops.github.io/pull/1942#discussion_r1796470529

I think "moonsettler" is an unusual string to appear here and could leave the reader wondering if they're missing something or we have a typo, so I add a title to provide context.  We don't usually provide titles for strings that are obviously names.

> - **CTV enhancement opcodes:** developer moonsettler posted to both

Reviewer: I did not recall this conversation. Makes sense.

Note to myself: make sure my first mention in Optech News is under pseudonym “born in 1830” so that the section starts with “developer born in 1830 posted to…” :smiling_imp:

> - **CTV enhancement opcodes:** developer moonsettler posted to both

Author: lol

  [Bitcoin-Dev][moonsettler ctvppml] and [Delving Bitcoin][moonsettler
  ctvppdelv] to propose two additional opcodes to be used in conjunction
  with the already-proposed [OP_CHECKTEMPLATEVERIFY][topic
  op_checktemplateverify]:

  - _OP_TEMPLATEHASH_ takes a list of transaction elements and converts
    it into a CTV-compatible hash.  This allows stack operations to
    specify details about the inputs to be used, the number of inputs,
    the locktime, the value of each output, the script of each output,
    the number of outputs, and the transaction version to use.

  - _OP_INPUTAMOUNTS_ puts the satoshi value of some or all of the

>     inputs on the stack, which it can be used as a parameter for
>     `OP_TEMPLATEHASH` (e.g. to require an equal output value).

Reviewer: I think both of these two suggestions would work:

```suggestion
    inputs on the stack, which it can use as a parameter for
    `OP_TEMPLATEHASH` (e.g. to require an equal output value).
```

Reviewer: ```suggestion
    inputs on the stack, which can be used as a parameter for
    `OP_TEMPLATEHASH` (e.g. to require an equal output value).
```

  the [time warp attack][topic time warp] by forbidding the first block
  in a new difficulty period from having a time more than 600 seconds
  earlier than the last block in the previous period.  Provoost is
  concerned that a honest miner using software that expands its nonce
  space by using a range of timestamps (_time rolling_) will
  accidentally produce a block that nodes with slow clocks might not
  immediately accept, slowing the block's propagation compared to
  competing blocks that may be produced around the same time that use
  less variable timestamps.  If a competing block remains on the best
  blockchain, the miner of the time-rolling block will lose income.
  Provoost instead suggests a less constrictive limit, such as
  forbidding time from running backwards more than 7,200 seconds (two
  hours).  Antoine Poinsot [defends][poinsot timewarp] the choice of 600
  seconds as avoiding any known problem and providing the strongest
  defense against future time warping.

## Releases and release candidates

_New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates._

> - [LND 0.18.4-beta.rc2][] is a release candidate for a minor version of

Reviewer 2: ```suggestion
- [LND v0.18.4-beta][] is a minor release of
```

  this popular LN implementation.

> - [Bitcoin Core 28.1RC1][] is a release candidate for a maintenance

Reviewer 2: ```suggestion
- [Bitcoin Core 28.1rc2][] is a release candidate for a maintenance
```

- [Bitcoin Core #31223][] changes how a node derives its “[Tor][topic anonymity
  networks] service target” P2P port (see Newsletter [#118][news118 tor]), using
  the user-specified `-port` value plus one instead of the default of 8334, if a
  `-port` value is given. This fixes a problem where multiple local nodes would
  all bind to 8334 and crash, causing port collusion. However, in the rare case
  where two local nodes are assigned consecutive `-port` values, they may still
  collide on the derived onion port, but this should be avoided altogether.

- [Eclair #2888][] implements support for the [peer storage][topic peer storage]
  protocol as specified in [BOLTs #1110][], which allows nodes to store
  encrypted backups for peers that request them, up to 65kB by default. This
  feature is intended for Lightning Service Providers (LSPs) serving mobile
  wallets, and has configurable settings that allow a node operator to specify
  how long it will store the data. This makes Eclair the second implementation
  to support peer storage, after CLN (see Newsletter [#238][news238 storage]).

- [LDK #3495][] refines the historical success probability scoring model in LN

>   pathfinding by improving the [probability density function[(PDF)][probability

Reviewer: ```suggestion
  pathfinding by improving the [probability density function (PDF)][probability
```

  real-world behavior, enhances default penalties, and improves pathfinding
  reliability.

- [LDK #3436][] moves the `lightning-liquidity` crate, which provides types and
  primitives to integrate an LSP (as specified [here][lsp spec]) with an
  LDK-based node, to the `rust-lightning` repository.

- [LDK #3435][] adds an authentication field to the [blinded path][topic rv
  routing] payment context message for a payer to include a Hash-based Message
  Authentication Codes (HMAC) and a nonce and allow a recipient to authenticate
  the payer. This fixes an issue where an attacker could take a `payment_secret`
  from a [BOLT11][] invoice issued by the victim node and forge a payment, even
  if it didn't match the amount expected for that [offer][topic offers]. This
  also helps prevent deanonymization attacks using the same technique.

- [LDK #3365][] ensures that the `holder_commitment_point` (next commitment
  point) is immediately marked as available on upgrades by retrieving it in
  `get_channel_reestablish`, instead of leaving it in the previously used
  `PendingNext` state. This change prevents forced closures when a channel is in

>   a steady state during an upgrade and then receives a `commitment_signed
>   message` that requires the next commitment point to be available.

Reviewer: ```suggestion
  a steady state during an upgrade and then receives a `commitment_signed`
  message that requires the next commitment point to be available.
```

---
title: 'Bitcoin Optech Newsletter #335'
permalink: /en/newsletters/2025/01/03/
name: 2025-01-03-newsletter
slug: 2025-01-03-newsletter
type: newsletter
layout: newsletter
lang: en
---

> This week's links to information about longstanding deanonymization

Reviewer: ```suggestion
This week's newsletter links to information about longstanding deanonymization
```

  keep their proposal in sync with his proposal for Bitcoin-compatible
  FROST signing (see [Newsletter #315][news315 frost]).

## Changing consensus

_A new monthly section summarizing proposals and discussion about
changing Bitcoin's consensus rules._

- **CTV enhancement opcodes:** developer moonsettler posted to both
  [Bitcoin-Dev][moonsettler ctvppml] and [Delving Bitcoin][moonsettler
  ctvppdelv] to propose two additional opcodes to be used in conjunction
  with the already-proposed [OP_CHECKTEMPLATEVERIFY][topic
  op_checktemplateverify]:

  - _OP_TEMPLATEHASH_ takes a list of transaction elements and converts
    it into a CTV-compatible hash.  This allows stack operations to
    specify details about the inputs to be used, the number of inputs,
    the locktime, the value of each output, the script of each output,
    the number of outputs, and the transaction version to use.

  - _OP_INPUTAMOUNTS_ puts the satoshi value of some or all of the
    inputs on the stack, which can be used as a parameter for
    `OP_TEMPLATEHASH` (e.g. to require an equal output value).

  Together these opcodes can create [vaults][topic vaults] with similar

>   properties to those possible with [BIP345][]'s `OP_VAULT`.  The

Reviewer 3: Not quite. While this does allow spending multiple vaulted outputs together, it is missing two of `OP_VAULT` properties:

- No partial re-vaulting support, the vaulted outputs must be spent in full to the staging script.
- No dynamic construction of the destination tap leaf, which enables choosing the withdrawal destination address at trigger time. Vaults using moonsettler's proposal would be limited to using static pre-determined destination addresses, similarly to simple `OP_CTV`-only vaults.

  encrypted backups for peers that request them, up to 65kB by default. This
  feature is intended for Lightning Service Providers (LSPs) serving mobile
  wallets, and has configurable settings that allow a node operator to specify
  how long it will store the data. This makes Eclair the second implementation
  to support peer storage, after CLN (see Newsletter [#238][news238 storage]).

- [LDK #3495][] refines the historical success probability scoring model in LN
  pathfinding by improving the [probability density function (PDF)][probability
  density] and related parameters based on real-world data collected from
  randomized probes. This PR aligns historical and a priori models with
  real-world behavior, enhances default penalties, and improves pathfinding
  reliability.

- [LDK #3436][] moves the `lightning-liquidity` crate
  to the `rust-lightning` repository.  This crate provides types and
  primitives to integrate an LSP (as specified [here][lsp spec]) with an
  LDK-based node.

- [LDK #3435][] adds an authentication field to the [blinded path][topic rv
  routing] payment context message for a payer to include a Hash-based Message

>   Authentication Codes (HMAC) and a nonce and allow a recipient to authenticate

Reviewer 4: ```suggestion
  Authentication Code (HMAC) and a nonce and allow a recipient to authenticate
```
